Nos/Controller/NoteEditorController.swift

```diff
@@ -0,0 +1,256 @@
+        if handledChange {
+            return false
+        }
+        textView.typingAttributes = defaultNSAttributes
+        
+        if text == "@" {
+            let showedAutocomplete = checkForMentionsAutocomplete(in: textView, at: nsRange)
+            if showedAutocomplete {
+                return true
+            }
+        } else if text.count > 1 {
+            let insertedNostrLink = handleNostrIdentifiers(in: text, textView: textView)
+            if insertedNostrLink {
+                return false
+            }
+        } 
+        
+        return true
+    }
+
+    // MARK: - Helpers 
+    
+    /// Calculates the height that fits all entered text, and updates $intrinsicHeight with this property. This is 
+    /// done to work around some incompatibilities between UIKit and SwiftUI where the UITextView won't expand properly.
+    func updateIntrinsicHeight(view: UIView) {
+        let newSize = view.sizeThatFits(CGSize(width: view.frame.width, height: .greatestFiniteMagnitude))
+        guard intrinsicHeight != newSize.height else { return }
+        DispatchQueue.main.async { // call in next render cycle.
+            self.intrinsicHeight = newSize.height
+        }
+    }
+    
+    /// Inserts the name of an author at the given range with a nostr link attached as an attribute. This function
+    /// assumes the cursor is directly after an '@' and will replace that too.
+    private func insertMention(of author: Author, at range: NSRange) {
+        guard let textView, let url = author.uri else {
+            return
+        }
+        
+        let rangeIncludingAmpersand = NSRange(
+            location: max(range.location - 1, 0), 
+            length: min(range.length + 1, textView.attributedText.length)
+        )
+        insert(text: "@\(author.safeName)", link: url.absoluteString, at: rangeIncludingAmpersand)
+    }
+    
+    /// Handles the user pasting an npub at the given range.
+    private func insertMention(npub: String, range: NSRange) {
+        insert(text: npub.prefix(10).appending("..."), link: "nostr:\(npub)", at: range)
+    }
+    
+    /// Handles the user pasting an note link at the given range. 
+    private func insertMention(note: String, range: NSRange) {
+        insert(text: note.prefix(10).appending("..."), link: "nostr:\(note)", at: range)
+    }
+    
+    /// Inserts a string at the given range and adds the link attribute with the given `link`.
+    private func insert(text: String, link: String, at range: NSRange) {
+        guard let textView else {
+            return
+        }
+        
+        let linkAttributes = defaultNSAttributes.merging(
+            [NSAttributedString.Key.link: link], 
+            uniquingKeysWith: { _, key in key }
+        )
+        let link = NSAttributedString(
+            string: text,
+            attributes: linkAttributes
+        )
+        
+        let attributedString = textView.attributedText.mutableCopy() as! NSMutableAttributedString
```

Comment:
I'm surprised SwiftLint isn't flagging this force cast. Did you consider any alternatives to this?

Reply:
oh yeah, I meant to get rid of these. I put them into guards now, and I enabled `force_cast` warnings for swiftlint, there actually weren't any others in the code.

Reply:
CC @martindsq for changing the swiftlint rules but I'm pretty sure you'll be good with this one :)